src/v/config/rest_authn_endpoint.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "config/convert.h"
+#include "config/from_string_view.h"
+#include "config/property.h"
+#include "json/_include_first.h"
+#include "json/stringbuffer.h"
+#include "json/writer.h"
+#include "net/unresolved_address.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <yaml-cpp/node/node.h>
+
+#include <iosfwd>
+#include <optional>
+#include <string>
+
+namespace config {
+
+enum class rest_authn_type {
+    none = 0,
+    mtls,
```

Comment:
>Since mtls is essentially none (the configuration and validation happen in the tls config section), does having it help or hinder?

I think mTLS is unnecessary here for your exact reason but some RFC reviews thought it would be more cohesive. I'm OK removing it though.

> If mtls is specified here, should there be a validation step to ensure that require_client_auth: true?
That makes sense. I'll add it in if we want to keep mTLS as an option.

Reply:
I'm going to remove mtls for now after @BenPope and I found a nullptr issue with using TLS certs on Pandaproxy.
I need to create a ticket for that issue as well.